rollup/pipeline/pipeline.go

```diff
@@ -103,22 +112,20 @@ func (p *Pipeline) WithBeforeTxHook(beforeTxHook func()) *Pipeline {
 func (p *Pipeline) Start(deadline time.Time) error {
 	p.start = time.Now()
 	p.txnQueue = make(chan *types.Transaction)
-	applyStageRespCh, candidateCh, err := p.traceAndApplyStage(p.txnQueue)
+	applyStageRespCh, applyDownstreamCh, err := p.traceAndApplyStage(p.txnQueue)
 	if err != nil {
 		log.Error("Failed starting traceAndApplyStage", "err", err)
 		return err
 	}
 	p.applyStageRespCh = applyStageRespCh
-	p.ResultCh = p.cccStage(candidateCh, deadline)
+	encodeDownstreamCh := p.encodeStage(applyDownstreamCh)
+	p.ResultCh = p.cccStage(encodeDownstreamCh, deadline)
 	return nil
 }
 
 // Stop forces pipeline to stop its operation and return whatever progress it has so far
 func (p *Pipeline) Stop() {
-	if p.txnQueue != nil {
-		close(p.txnQueue)
-		p.txnQueue = nil
-	}
+	close(p.txnQueue)
```

Comment:
Any specific reason you changed this part?

Reply:
I guess it should never be nil if `Pipeline` is used correctly (call start, process pipeline, stop in the correct order

Reply:
Correct. With Stop and Release closing different channels now, as long as caller uses Pipeline as expected, double close should never be an issue.